pandas/io/excel/_base.py

```diff
@@ -1617,11 +1617,3 @@ def __exit__(
         traceback: TracebackType | None,
     ) -> None:
         self.close()
-
-    def __del__(self) -> None:
```

Comment:
Why did you delete this?

Reply:
IMO we shouldn't rely on garbage collection to perform a final closure of files from `ExcelWriter`. It didn't work correctly from the `ResourceWarning`s issues picked up from this PR, and would be a band-aid fix at best

Reply:
thx